apps/src/applab/ChartApi.js

```diff
@@ -4,327 +4,284 @@
    */
-  this.warnings = [];
-}
+  constructor(docContext, appStorage) {
+    this.document_ = docContext || document;
+    this.appStorage_ = appStorage || Applab.storage;
 
-/**
- * Record a runtime warning.
- * @param {string} warningMessage
- */
-ChartApi.prototype.warn = function(warningMessage) {
-  this.warnings.push(new Error(warningMessage));
-};
+    /**
+     * List of all warnings generated while performing operations through this
+     * API instance.
+     * @type {Error[]}
+     */
+    this.warnings = [];
+  }
 
-/**
- * Add warnings from an array to the ChartApi instance's warnings array.
- * @param {Error[]} newWarnings
- * @private
- */
-ChartApi.prototype.mergeWarnings_ = function(newWarnings) {
-  Array.prototype.push.apply(this.warnings, newWarnings);
-};
+  /**
+   * Record a runtime warning.
+   * @param {string} warningMessage
+   */
+  warn(warningMessage) {
+    this.warnings.push(new Error(warningMessage));
+  }
 
-// When adding a new type, provide an entry in the string enum (for clean code)
-// and an entry in the TypeNameToType map (allows us to easily remap different
-// implementations to the same type name).
+  /**
+   * Add warnings from an array to the ChartApi instance's warnings array.
+   * @param {Error[]} newWarnings
+   * @private
+   */
+  mergeWarnings_(newWarnings) {
+    Array.prototype.push.apply(this.warnings, newWarnings);
+  }
 
-/** @enum {string} */
-ChartApi.ChartType = {
-  BAR: 'bar',
-  PIE: 'pie',
-  LINE: 'line',
-  SCATTER: 'scatter'
-};
+  // When adding a new type, provide an entry in the string enum (for clean code)
+  // and an entry in the TypeNameToType map (allows us to easily remap different
+  // implementations to the same type name).
 
-/** @type {Object.<string, GoogleChart>} */
-ChartApi.TypeNameToType = {
-  bar: GoogleChart.MaterialBarChart,
-  pie: GoogleChart.PieChart,
-  line: GoogleChart.MaterialLineChart,
-  scatter: GoogleChart.MaterialScatterChart
-};
+  /** @enum {string} */
+  static ChartType = {
+    BAR: 'bar',
+    PIE: 'pie',
+    LINE: 'line',
+    SCATTER: 'scatter'
+  };
 
-/**
- * Get an array of all the chart type strings.
- * @returns {string[]}
- */
-ChartApi.getChartTypeNames = function() {
-  return Object.getOwnPropertyNames(ChartApi.TypeNameToType);
-};
+  /** @type {Object.<string, GoogleChart>} */
+  static TypeNameToType = {
+    bar: GoogleChart.MaterialBarChart,
+    pie: GoogleChart.PieChart,
+    line: GoogleChart.MaterialLineChart,
+    scatter: GoogleChart.MaterialScatterChart
+  };
 
-/**
- * @param {ChartType} chartType
- * @returns {boolean} TRUE if the given type is in the known list of chart types.
- */
-ChartApi.supportsType = function(chartType) {
-  return ChartApi.getChartTypeNames().indexOf(chartType.toLowerCase()) !== -1;
-};
+  /**
+   * Get an array of all the chart type strings.
+   * @returns {string[]}
+   */
+  static getChartTypeNames() {
+    return Object.getOwnPropertyNames(ChartApi.TypeNameToType);
```

Comment:
Why `Object.getOwnPropertyNames` vs. `Object.keys`?

Reply:
Because when this was written [in September 2015](https://github.com/code-dot-org/code-dot-org/commit/5427b44035f0b12d4cb2460db3f9fa1f44113ef3) I didn't know any better. :joy: